complex/ft_main.c

```diff
@@ -321,20 +321,40 @@ static int ft_fw_server(void)
 	return ret;
 }
 
+static int ft_provider_check(const char *req_name, const char *fab_name)
+{
+	const char *comp;
+
+	if (!strcmp(req_name, fab_name))
+		return 0;
+
+	comp = strchr(fab_name, ';');
+	if (comp)
+		return strcmp(req_name, ++comp);
+
+	return 1;
+}
+
```

Comment:
I would suggest to use ft_core_name, ft_util_name functions here. They have comprehensive checks.